src/main/resources/com/owlcyberdefense/mil-std-2045/xsd/milstd2045_application_header.dfdl.xsd

```diff
@@ -0,0 +1,628 @@
+                          <sequence dfdl:hiddenGroupRef="msi:message_standard_version_PI"/>
+                          <element name="message_standard_version"
+                                   minOccurs="0"
+                                   dfdl:occursCountKind="expression"
+                                   dfdl:occursCount='{
+                                  if (../umf/value eq "OK_Binary_File" or
+                                      ../umf/value eq "OK_RDM" or
+                                      fn:starts-with(../umf/value, "Undefined") )
+                                      then 0
+                                      else ../message_standard_version_PI/value
+                                }'>
+                            <complexType>
+                              <choice dfdl:choiceDispatchKey="{ ../umf/value }">
+                                <element dfdl:choiceBranchKey="OK_Link_16" name="link16"
+                                         type="tns:messsageStandardVersionLink16Enum"/>
+                                <element dfdl:choiceBranchKey="OK_VMF" name="vmf"
+                                         type="tns:messsageStandardVersionVMFEnum"/>
+                                <element dfdl:choiceBranchKey="OK_NITFS" name="nitfs"
+                                         type="tns:messsageStandardVersionNITFSEnum"/>
+                                <element dfdl:choiceBranchKey="OK_XML-MTF" name="xml-mtf"
+                                         type="tns:messsageStandardVersionXML_MTFEnum"/>
+                                <element dfdl:choiceBranchKey="OK_XML-VMF" name="xml-vmf"
+                                         type="tns:messsageStandardVersionXML_VMFEnum"/>
+                              </choice>
+                            </complexType>
+                          </element>
+                          <sequence dfdl:hiddenGroupRef="msi:vmf_message_identification_group_PI"/>
+                          <element name="vmf_message_identification_group"
+                                   minOccurs="0"
+                                   dfdl:occursCountKind="expression"
+                                   dfdl:occursCount="{ ../vmf_message_identification_group_PI/value }">
+                            <complexType>
+                              <sequence>
+                                <element name="fad">
+                                  <complexType>
+                                    <sequence>
+                                      <element name="value" type="tns:tIntField"
+                                               dfdl:length="4"/>
+                                      <element name="functional_area" type="tns:FADCodewordsEnum"
+                                               dfdl:inputValueCalc='{
+                                  if (../value eq 0) then "OK_Network_Control"
+                                  else if (../value eq 1) then "OK_General_Information_Exchange"
```

Comment:
Is there any particular reason this is a IVC rather than an enum?

Reply:
Yes, because we need both the value as a numeric element in the infoset AND we want to compute an enum string from it for readability. 

But I think I will simplify and get rid of this FADCodeWordEnum entirely.